### PR TITLE
DEC-443 display errors

### PR DESCRIPTION
**Pull request by danthewolfe**
* Created a mediator for sending messages to subscribers. Subscribers provide a specific call back function that will be called when a message that they are concerned with is received.
 * Cleaned up the different form messages since they were duplicating code for the most part, aside from the message and style
 * Created a FormMessageCenter that handles displaying all "global" messages to the user. This object subscribes to certain messages for triggering the display of a message, focusing on the message, and hiding it.
 * Created an APIResponseMixin to handle parsing the response json when errors are received from the api to get an easier to use error message
 * Stripped out the error/success messaging parts of ItemMetaDataForm and changed it to utilize the message center object for this
 * Modified CollectionPublishToggle to display an error message when it is unable to publish/unpublish the collection.

======

 * Added and implemented material-ui for the error dialog.
 * Created some mixins for general material-ui theme related stuff and for dialog components.

### CODE REVIEW

File: app/assets/javascripts/mediator.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+function Mediator() {
+  // hashmap of message type to list of subscribers for that message type
+  this.subscribers = [];
+};
+
+Mediator.prototype.subscribe = function(type, subscriberCallback) {
+  if (!(type in this.subscribers))
+    this.subscribers[type] = [];
```
Comment (jaronkk) [inline app/assets/javascripts/mediator.js]: This if statement needs curly braces

File: app/assets/javascripts/components/forms/FormMessageCenter.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  receiveDisplay: function(type, message) {
+    this.setState({
+      messageType: message[0],
+      messageText: message[1],
+    });
+    this.refs.errorDialog.show();
+  },
+
+  receiveDisplayAndFocus: function(type, message) {
```
Comment (jgondron) [inline app/assets/javascripts/components/forms/FormMessageCenter.jsx]: This is no longer needed now that we're using the dialogs as a way to ensure the user sees the message.

**Comment by rfox2** [comment]
:+1: Would the mediator code need to be deprecated if we switch to flux?

**Comment by danthewolfe** [comment]
Probably. I think we'd remove the mediator.js file and use flux's dispatcher instead. It would still be a very similar pattern of subscribing and event emitter.

**Pull request merged**